test/as_buffer_async_pubsub_1.cpp

```diff
@@ -76,13 +76,12 @@ BOOST_AUTO_TEST_CASE( pub_qos0_sub_qos0 ) {
                     BOOST_TEST(packet_id == pid_sub);
                     BOOST_TEST(results.size() == 1U);
                     BOOST_TEST(*results[0] == mqtt::qos::at_most_once);
-                    auto topic1 = std::make_shared<std::string>("topic1");
-                    auto contents = std::make_shared<std::string>("topic1_contents");
                     c->async_publish_at_most_once(
-                        as::buffer(*topic1),
-                        as::buffer(*contents),
-                        false,
-                        [topic1, contents](boost::system::error_code const&) {}
+                        as::buffer("topic1"),
```

Comment:
I don't want to add '\0' after the string. Please use https://github.com/redboltz/mqtt_cpp/pull/286/files#diff-c039c862d67b802e139cb035c26dcc4dR235 way.

Reply:
See https://github.com/redboltz/mqtt_cpp/issues/277#issuecomment-501517510

Reply:
And please do the same update other places on the test.